### PR TITLE
refactor: Remove unreliable info from txn meta

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
Implements https://app.clickup.com/t/9002019994/PRF-6186

Redefines the contract data categorization handling for EVM transactions. The changeset removes the unreliable information from the transaction metadata and instead drops it or infers it from the transaction itself.

Changes:
- Remove `evm_is_token_whitelisted` global variable
- Replace `PAYLOAD_STATUS` enum with `EVM_TRANSACTION_TYPE`
- Remove the token symbol from metadata and infer it from the whitelisted token list
- Update user confirmation prompts for token
- Remove the contract list from the app config and only provide specific contract references

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-6186 Remove unreliable txn meta information](https://app.clickup.com/t/9002019994/PRF-6186)

File: apps/evm_family/evm_txn_helpers.c [hunk 1]
```diff
@@ -174,56 +168,49 @@ static uint64_t get_decode_length(const uint8_t *seq,
...
+      EVM_deposit_TAG == function_tag || EVM_transfer_TAG == function_tag) {
+    // decode the contract data for display
+    if (ETH_UTXN_ABI_DECODE_OK !=
+        ETH_ExtractArguments(txn_context->transaction_info.data,
+                             txn_context->transaction_info.data_size,
+                             &txn_context->display_node)) {
+      return EVM_TXN_INVALID_DATA;
+    } else {
+      return EVM_TXN_KNOWN_FUNC_SIG;
     }
```
Comment (amanCypherock) [inline apps/evm_family/evm_txn_helpers.c:200]: Add a todo to prevent collision of function tags

File: apps/evm_family/evm_txn_helpers.c [hunk 2]
```diff
@@ -174,56 +168,49 @@ static uint64_t get_decode_length(const uint8_t *seq,
...
+
+  if (EVM_swap_TAG == function_tag || EVM_uniswapV3Swap_TAG == function_tag ||
+      EVM_safeTransferFrom_TAG == function_tag ||
+      EVM_deposit_TAG == function_tag || EVM_transfer_TAG == function_tag) {
+    // decode the contract data for display
+    if (ETH_UTXN_ABI_DECODE_OK !=
+        ETH_ExtractArguments(txn_context->transaction_info.data,
+                             txn_context->transaction_info.data_size,
+                             &txn_context->display_node)) {
+      return EVM_TXN_INVALID_DATA;
```
Comment (amanCypherock) [inline apps/evm_family/evm_txn_helpers.c:197]: Please add a comment here for which condition would be an Invalid case.

File: apps/evm_family/evm_user_verification.c [hunk 3]
```diff
@@ -122,22 +119,24 @@ bool evm_verify_transfer(const evm_txn_context_t *txn_context) {
 #endif
 
   // verify recipient address; TODO: handle harmony address encoding
-  eth_get_to_address(&txn_context->transaction_info, &to_address);
+  eth_get_to_address(txn_context, &to_address);
   ethereum_address_checksum(
       to_address, &address[2], false, g_evm_app->chain_id);
-  if (!core_scroll_page(title, address, evm_send_error)) {
+  snprintf(
+      display, sizeof(display), UI_TEXT_BTC_SEND_PROMPT, unit, g_evm_app->name);
```
Comment (amanCypherock) [inline apps/evm_family/evm_user_verification.c:126]: Either rename the macro or create a new one for this

File: apps/evm_family/evm_txn_helpers.h [hunk 4]
```diff
@@ -30,23 +30,22 @@
...
+  EVM_TXN_NO_DATA = 0x0,
+
+  /// Data present in the transaction is invalid
+  EVM_TXN_INVALID_DATA,
+
+  /// Function signature in data is not recognized
+  EVM_TXN_UNKNOWN_FUNC_SIG,
+
+  /// Data is for token transfer function
+  EVM_TXN_TRANSFER_FUNC,
```
Comment (amanCypherock) [inline apps/evm_family/evm_txn_helpers.h:44]: Let's consider the name EVM_TXN_TOKEN_TRANSFER_FUNC

File: apps/evm_family/evm_txn_helpers.c (hunk 2 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_txn_helpers.c:203]: added

File: apps/evm_family/evm_user_verification.c (hunk 3 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_user_verification.c]: renamed

File: apps/evm_family/evm_txn_helpers.h (hunk 4 above)
Comment (ujjwal-cyph) [reply apps/evm_family/evm_txn_helpers.h]: renamed

**Review by amanCypherock — APPROVED** [review]

**Pull request merged**